src/darwin/Framework/CHIPTests/TestHelpers/MTRTestCase.mm

```diff
@@ -106,6 +122,22 @@ - (void)tearDown
     [super tearDown];
 }
 
++ (id)createControllerOnTestFabric
+{
+    __auto_type * storage = [[MTRTestStorage alloc] init];
+    __auto_type * factoryParams = [[MTRDeviceControllerFactoryParams alloc] initWithStorage:storage];
+    __auto_type * factory = MTRDeviceControllerFactory.sharedInstance;
+    XCTAssertTrue([factory startControllerFactory:factoryParams error:nil]);
+
+    __auto_type * testKeys = [[MTRTestKeys alloc] init];
+    __auto_type * params = [[MTRDeviceControllerStartupParams alloc] initWithIPK:testKeys.ipk fabricID:@1 nocSigner:testKeys];
+    params.vendorID = @0xFFF1;
+    MTRDeviceController * controller = [factory createControllerOnNewFabric:params error:nil];
```

Comment:
I know you copied this, but followup: we should probably change this to use the new controller-creation APIs.